Modules/Module_Processing/xASL_init_DefaultEffectiveResolution.m

```diff
@@ -30,14 +30,39 @@
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % REFERENCES: Petr, 2018 MAGMA; Vidorreta 2013 Neuroimage
 %
-% Copyright 2015-2020 ExploreASL
+% Copyright 2015-2024 ExploreASL
 
 %% ----------------------------------------------------------------------------------------
 %% Admin
 tIM = xASL_io_ReadNifti(PathASL);
+[fPath, fName] = xASL_fileparts(PathASL);
+
+% Read JSON of the ASL scan
+tJson = xASL_io_ReadJson([fullfile(fPath, fName) '.json']);
+tJson = xASL_bids_parms2BIDS([], tJson, 0);
+
+% Obtain the resolution based on the image matrix
 NativeResolution = tIM.hdr.pixdim(2:4);
-if ~isfield(x.Q,'Sequence')
-    warning(['Setting x.Q.Sequence missing, skipping']);
+
+% Load the x.Q fields from the JSON
```

Comment:
This is nice but I would add this to 

xASL_bids_parms2BIDS

Because PulseSequenceType -> Sequence

inside xASL_bids_parms2BIDS (which may not be correct) and there is more flexibility to do this list in parms2BIDS rather than only here. 

Here we initialize the default effective resolution. We don't translate parms2BIDS (sequence, readoutdim, vendor, is all translation).